test/integ/test_connection.py

```diff
@@ -1107,3 +1108,73 @@ def test_ocsp_cache_working(conn_cnx):
     with conn_cnx() as cnx:
         assert cnx
     assert OCSP_CACHE.telemetry["hit"] + OCSP_CACHE.telemetry["miss"] > original_count
+
+
+@pytest.mark.skipolddriver
+def test_imported_packages_telemetry(conn_cnx, capture_sf_telemetry, db_parameters):
+    try:
+        # these imports are not used but for testing
+        import html.parser  # noqa: F401
+        import json  # noqa: F401
+        import multiprocessing as mp  # noqa: F401
+        from datetime import date  # noqa: F401
+        from math import sqrt  # noqa: F401
+
+        with conn_cnx() as conn, capture_sf_telemetry.patch_connection(
+            conn, False
+        ) as telemetry_test:
+            conn._log_telemetry_imported_packages()
+            assert len(telemetry_test.records) > 0
+            for t in telemetry_test.records:
+                if (
+                    t.message[TelemetryField.KEY_TYPE.value]
+                    == TelemetryField.IMPORTED_PACKAGES.value
+                ):
+                    assert "pytest" in t.message["value"]
+                    assert "unittest" in t.message["value"]
+                    assert "json" in t.message["value"]
+                    assert "multiprocessing" in t.message["value"]
+                    assert "html" in t.message["value"]
+                    assert "datetime" in t.message["value"]
+                    assert "math" in t.message["value"]
+                    assert "__main__" not in t.message["value"]
+                    assert CLIENT_NAME == t.message[TelemetryField.KEY_SOURCE.value]
+
+        # test different application
+        new_application_name = "PythonSnowpark"
+        config = {
+            "user": db_parameters["user"],
+            "password": db_parameters["password"],
+            "host": db_parameters["host"],
+            "port": db_parameters["port"],
+            "account": db_parameters["account"],
+            "schema": db_parameters["schema"],
+            "database": db_parameters["database"],
+            "protocol": db_parameters["protocol"],
+            "timezone": "UTC",
+            "application": new_application_name,
+        }
+        with snowflake.connector.connect(
+            **config
+        ) as conn, capture_sf_telemetry.patch_connection(conn, False) as telemetry_test:
+            conn._log_telemetry_imported_packages()
+            assert len(telemetry_test.records) > 0
+            for t in telemetry_test.records:
+                if (
+                    t.message[TelemetryField.KEY_TYPE.value]
+                    == TelemetryField.IMPORTED_PACKAGES.value
+                ):
+                    assert (
+                        new_application_name
+                        == t.message[TelemetryField.KEY_SOURCE.value]
+                    )
```

Comment:
I'm a fan of using a generator of booleans and `any` in cases of this. If the telemetry changes then the test in its current form would silently not test anything. With `any` we make sure that if occurs once.
```suggestion
            any(
                [
                    t.message[TelemetryField.KEY_TYPE.value] == TelemetryField.IMPORTED_PACKAGES.value
                    and new_application_name == t.message[TelemetryField.KEY_SOURCE.value]
                    for t in telemetry_test.records:
                ]
            )
```

Reply:
done